src/cryptsetup/cryptsetup-generator.c

```diff
@@ -238,6 +238,7 @@ static int create_disk(
         tmp = fstab_test_option(options, "tmp\0");
         swap = fstab_test_option(options, "swap\0");
         netdev = fstab_test_option(options, "_netdev\0");
+        create_in_initrd = fstab_test_option(options, "x-initrd.create\0");
```

Comment:
This would work, but I think we should be able to make that automatic. Anything that is a dependency of `-.mount` should be excluded from killing. This makes yet another bit of internal state visible and configurable from the outside.

What about the following: on reload, create a drop-in that cancels the Conflicts=shutdown.target setting for the device configured as backing for `-.mount`? I'm not sure about the best implementation, but maybe create
the conflicts in a special `dev-<foo>.d/10-default-shutdown.conf` from cryptsetup target, and override that with `dev-<foo>.d/20-default-shutdown.conf` from any other target that configures `-.mount` or `usr.mount`, etc.

Reply:
the problem with these automatic deps is that DM devices have multiple names and /proc/self/mountinfo is likely going to show a generic one (i.e. /dev/md-0 rather than /dev/mapper/luks-b5d403b9-344c-43e9-9fbc-daa8596f204c), hence we are likely to not match things up correctly. I fear tagging something manually is something we cannot avoid... (or to turn this arround: maybe add the automatic mechanism too, but the explicit one like @fbuihuu implemented it in this PR is something to have as well, already because people might do crazy stuff like splitting of /usr/lib/share and thus might need to mark more stuff as "mount in initrd")

Reply:
@keszybz I agree with @poettering: even if we could make it automatic for / and /usr we still need the option to deal with other paths that need to be mounted from initrd (and yes we already have some use cases here with transactional systems).

That said I'm not sure to understand your proposal. The drop-ins adding conflicts would be added during reload, which makes the whole thing depends on the fact that a reload is done before rebooting. It's very likely indeed but still seems a bit fragile no ?

Also shouldn't the conflicts have to be added to the systemd-cryptsetup services ?